enumfields/fields.py

```diff
@@ -31,7 +32,38 @@ def __set__(self, obj, value):
         obj.__dict__[self.field.name] = self.field.to_python(value)
 
 
-class EnumFieldMixin:
+class EnumFieldValidationMixin:
+
+    def _raise_error_if_next_value_not_allowed(self, value, model_instance):
+        if model_instance.pk is not None:
+            previous_values = [choice.value for choice in self.enum.get_allowed_previous_choices(value.name)]
+
+            # search whether in db exists record within all posible previous states and the current one
+            if not self.model.objects.filter(
+                Q(**{'{}__in'.format(self.name): previous_values + [value.value]}) | Q(**{self.name: None}),
+                pk=model_instance.pk,
+            ).exists():
```

Comment:
to get rid of this db hit we would need to add dependency on https://github.com/druids/django-chamber/ and declare the fields must be used only with SmartModel. otherwise I cannot get the changed fields...

Reply:
I don't like it, better solution will be to store prev enum value during model inicialization (get model from db) to some cache or new db field. There is problem with more transactions and isolation level.

Reply:
Why not just get current value of the field from DB, transform it to enum member, and compare it's next attribute with the value being set?

Reply:
because you work with the instance from original state in the time you loaded it from the DB.

Reply:
OK, but I have current value as the `value` argument right? So I can get the value from DB and check if current value is in its `next` property, right?

Reply:
Now I am caching the original value and the condition changed. No db access this time.

Reply:
@rubickcz no you shouldn't check value from the database. Because you can have inconsistent request flow with your solution.